### PR TITLE
NOT RELEVANT Making modifications to pass CLI tests.

### PR DESCRIPTION
**Pull request by Jad-yehya**
Changing `torch` dependency and not using `pip`.
Creating the subfolder for the data if it does not already exist in datasets.

### CODE REVIEW

File: datasets/psm.py [hunk 1]
```diff
@@ -28,16 +28,19 @@ class Dataset(BaseDataset):
     }
 
     def get_data(self):
-        # Check if the data is already here
         path = config.get_data_path(key="PSM")
-
+        # Check if the data is already here
         if not pathlib.Path.exists(path):
```
Comment (tomMoral) [inline datasets/psm.py]: `path.exists()`?

File: solvers/AR.py [hunk 2]
```diff
@@ -139,6 +146,8 @@ def run(self, _):
 
         if xw_hat.is_cuda:
             xw_hat = xw_hat.detach().cpu().numpy()
+        else:
+            xw_hat = xw_hat.detach().numpy()
```
Comment (tomMoral) [inline solvers/AR.py]: You can always call `cpu` even if it is already on CPU so remove the if.

File: solvers/lstm.py [hunk 3]
```diff
@@ -70,11 +37,49 @@ def prepare_data(self, *data):
             for d in data)
 
     def set_objective(self, X_train, y_test, X_test):
+        class LSTM_Autoencoder(torch.nn.Module):
```
Comment (tomMoral) [inline solvers/lstm.py:41]: maybe put these models (`AR, LSTM`) in `benchmark_utils.models`?

Comment (Jad-yehya) [reply solvers/lstm.py:41]: Yes I will do this, I saw that the problem came from using torch outside of a function in the Solver class. I will put the models in `benchmark_utils.models` for cleaner code

**Pull request closed without merging**